### DISCUSSION TITLE
Hitting rate limit

### DISCUSSION THREAD

**Discussion by Fluubah** [Rapptz/discord.py#6118] [Q&A]
How can I prevent hitting the rate limit?
My bot uses a lot of stuff with reactions, so it's easy for it to hit the limit, But how do I prevent it from hitting the limit?
It got temp blocked because it hit the limit to many times.
Edit:I am new to github, So I am sorry if this is improperly formatted

**Comment by Harmon758** [comment]
The library should handle most rate limits. What rate limit are you hitting?

**Reply by Fluubah** [reply]
429, to many requests one.

**Reply by Harmon758** [reply]
That would probably be the HTTP response status code, yes. What specific rate limit are you hitting?

**Reply by Fluubah** [reply]
how would I check?

**Reply by Fluubah** [reply]
This is what it says discord.errors.HTTPException: 429 Too Many Requests (error code: 0): You are being blocked from accessing our API temporarily due to exceeding our rate limits frequently. Please read our docs at https://discord.com/developers/docs/topics/rate-limits to prevent this moving forward.

**Reply by Harmon758** [reply]
What are you attempting to do when it errors? What's the full traceback?

**Reply by Fluubah** [reply]
I am just trying to run the bot
Edit:One of the reasons might be that when you react to some messages it does other commands, and people might be doing reactions to fast.

**Reply by Harmon758** [reply]
You usually get this restriction [after exceeding 10,000 invalid requests in 10 minutes (a sustained 16 to 17 requests per second)](https://discord.com/developers/docs/topics/rate-limits#invalid-request-limit).

**Reply by Fluubah** [reply]
Hmm... I wonder why this is happening?
